distribution/lib/Standard/Table/0.0.0-dev/src/Column.enso

```diff
@@ -2099,52 +2101,62 @@ type Column
+         value is `Report_Unmatched` which means that the user expects that two
+         columns should have the same amount of rows; if they do not, the
+         behaviour is the same as if it was set to `True` - i.e. the unmatched
+         rows are kept with `Nothing` values for the other column, but a
+         `Row_Count_Mismatch` problem is also reported.
+       - right_prefix: The prefix added to right column column names in case of
+         name conflict. See "Column Renaming" below for more information.
+       - on_problems: Specifies how to handle problems if they occur, reporting
+         them as warnings by default.
 
-             example_zip =
-                Examples.integer_column.zip Examples.text_column_1 [_, _]
-    zip self that:Column (function : Any -> Any -> Any) skip_nothing:Boolean=True (expected_value_type : Value_Type | Auto = Auto) -> Column ! Invalid_Value_Type =
-        new_fn = case skip_nothing of
-            True ->
-                x -> y ->
-                    if x.is_nothing then Nothing else
-                        if y.is_nothing then Nothing else
-                            function x y
-            False -> function
-        new_name = naming_helper.binary_operation_name "x" self that
-        vec = self.to_vector.zip on_problems=No_Wrap.Value that.to_vector new_fn
-        Column.from_vector new_name vec value_type=expected_value_type
+         - If the columns have different number of rows and `keep_unmatched` is
+           set to `Report_Unmatched`, the join will report `Row_Count_Mismatch`.
+
+       ? Column Renaming
+
+         If columns from the two columns have colliding names, a prefix (by
+         default `Right `) is added to the name of the column from the right
+         column. The left column remains unchanged. It is possible that the new
+         name will be in use, in this case it will be resolved using the normal
+         renaming strategy - adding subsequent ` 1`, ` 2` etc.
+
+       ? Row Ordering For In-Memory Columns
+
+         This operation requires a well-defined order of rows in the input
+         columns. In-memory columns rely on the ordering stemming directly from
+         their layout in memory. Database columns may not impose a deterministic
+         ordering. The existing ordering can be overridden by specifying
+         a different ordering using `Column.sort`. If no primary key was
+         defined nor any ordering was specified explicitly by the user, the
+         order of columns is undefined and the operation will fail, reporting a
+         `Undefined_Column_Order` problem and returning an empty table.
+
+       ? Row Ordering For Database Columns
+
+         The ordering of rows in the resulting table is not specified.
+    @keep_unmatched (make_single_choice [["True", "Boolean.True"], ["False", "Boolean.False"], ["Report", Report_Unmatched.to Meta.Type . qualified_name]])
```

Comment:
I think we should create a helper method for the tri-state `True|False|Report_Unmatched` widge. This way we could avoid having to redefine it in many places.